fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/ShowCreateMaterializedViewCommand.java

```diff
@@ -0,0 +1,157 @@
+import org.apache.doris.common.AnalysisException;
+import org.apache.doris.common.ErrorCode;
+import org.apache.doris.common.ErrorReport;
+import org.apache.doris.common.util.Util;
+import org.apache.doris.mysql.privilege.PrivPredicate;
+import org.apache.doris.nereids.trees.plans.PlanType;
+import org.apache.doris.nereids.trees.plans.commands.info.TableNameInfo;
+import org.apache.doris.nereids.trees.plans.visitor.PlanVisitor;
+import org.apache.doris.qe.ConnectContext;
+import org.apache.doris.qe.ShowResultSet;
+import org.apache.doris.qe.ShowResultSetMetaData;
+import org.apache.doris.qe.StmtExecutor;
+
+import com.google.common.collect.Lists;
+
+import java.util.List;
+
+/**
+ * Represents the command for SHOW CREATE MATERIALIZED VIEW.
+ */
+public class ShowCreateMaterializedViewCommand extends Command implements NoForward {
+    private static final ShowResultSetMetaData MATERIALIZED_VIEW_META_DATA =
+            ShowResultSetMetaData.builder()
+                    .addColumn(new Column("TableName", ScalarType.createVarchar(20)))
+                    .addColumn(new Column("ViewName", ScalarType.createVarchar(30)))
+                    .addColumn(new Column("CreateStmt", ScalarType.createVarchar(500)))
+                    .build();
+
+    private final String dbName;
+    private final String mvName;
+    private final String tableName;
+    private final String catalog;
+
+    public ShowCreateMaterializedViewCommand(String mvName, TableNameInfo tableNameInfo) {
+        super(PlanType.SHOW_CREATE_MATERIALIZED_VIEW_COMMAND);
+        this.tableName = tableNameInfo.getTbl();
+        this.dbName = tableNameInfo.getDb();
+        this.catalog = tableNameInfo.getCtl();
+        this.mvName = mvName;
+    }
+
+    private void validate(ConnectContext ctx) throws AnalysisException {
+        TableName tbl = new TableName(catalog, dbName, tableName);
+
+        Analyzer analyzer = new Analyzer(ctx.getEnv(), ctx);
+        tbl.analyze(analyzer);
+
+        // disallow external catalog
+        Util.prohibitExternalCatalog(tbl.getCtl(), this.getClass().getSimpleName());
+        if (!Env.getCurrentEnv().getAccessManager()
+                .checkTblPriv(ConnectContext.get(), tbl.getCtl(), tbl.getDb(), tbl.getTbl(),
+                        PrivPredicate.SHOW)) {
+            ErrorReport.reportAnalysisException(ErrorCode.ERR_TABLEACCESS_DENIED_ERROR, "SHOW CREATE MATERIALIZED",
+                    ConnectContext.get().getQualifiedUser(),
+                    ConnectContext.get().getRemoteIP(),
+                    tbl.toSql());
+        }
+    }
+
+    public String getDbName() {
+        return dbName;
+    }
+
+    public String getMvName() {
+        return mvName;
+    }
+
+    public String getTableName() {
+        return tableName;
+    }
+
+    public String getCatalog() {
+        return catalog;
+    }
+
+    @Override
+    public void run(ConnectContext ctx, StmtExecutor executor) throws Exception {
```

Comment:
better to take the legacy planner's code as reference
```
private void handleShowCreateMaterializedView() throws AnalysisException {
        List<List<String>> resultRowSet = new ArrayList<>();
        ShowCreateMaterializedViewStmt showStmt = (ShowCreateMaterializedViewStmt) stmt;
        Database db = Env.getCurrentInternalCatalog().getDbOrAnalysisException(showStmt.getTableName().getDb());
        Table table = db.getTableOrAnalysisException(showStmt.getTableName().getTbl());
        if (table instanceof OlapTable) {
            OlapTable baseTable = ((OlapTable) table);
            Long indexIdByName = baseTable.getIndexIdByName(showStmt.getMvName());
            if (indexIdByName != null) {
                MaterializedIndexMeta meta = baseTable.getIndexMetaByIndexId(indexIdByName);
                if (meta != null && meta.getDefineStmt() != null) {
                    String originStmt = meta.getDefineStmt().originStmt;
                    List<String> data = new ArrayList<>();
                    data.add(showStmt.getTableName().getTbl());
                    data.add(showStmt.getMvName());
                    data.add(originStmt);
                    resultRowSet.add(data);
                }
            }
        }
        resultSet = new ShowResultSet(showStmt.getMetaData(), resultRowSet);
    }
```
We just need some minor changes to migrate the old code to nereids, and it's more clear and easy to review. If rewrite the logical completely, it will take a lot more time to review the code and test it. And the pr may not be quickly merged until multiple round of test. So could you please re-implement these as legacy planner's logic

Reply:
Done